python/tchannel/sync/thrift.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) 2015 Uber Technologies, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from __future__ import absolute_import
+
+import inspect
+
+from tchannel.thrift.client import client_for as async_client_for
+
+
+def client_for(service, service_module, thrift_service_name=None):
+    """Build a sync client class for the given Thrift service.
+
+    The generated class accepts a TChannelSyncClient and an optional
+    hostport as initialization arguments.
+
+    Given ``CommentService`` defined in ``comment.thrift`` and registered
+    with Hyperbahn under the name "comment", here's how this might be used:
+
+    .. code-block:: python
+
+        from tchannel.sync import TChannelSyncClient
+        from tchannel.sync.thrift import client_for
+
+        from comment import CommentService
+
+        CommentServiceClient = client_for('comment', CommentService)
+
+        tchannel_sync = TChannelSyncClient('my-service')
+        comment_client = CommentServiceClient(tchannel_sync)
+
+        future = comment_client.postComment(
+            articleId,
+            CommendService.Comment("hi")
+        )
+        result = future.result()
+
+    :param service:
+        Name of the HyperBahn service being called.
+    :param service_module:
+        The Thrift-generated module for that service. This usually has
+        the same name as definied for the service in the IDL.
+    :param thrift_service_name:
+        If the Thrift service has a different name than its module, use
+        this parameter to specify it.
+    :returns:
+        An Thrift-like class, ready to be instantiated and used
+        with TChannelSyncClient.
```

Comment:
The docs need to mention somewhere that the methods will return futures where .result() will block until a result is available.